### DISCUSSION TITLE
Add TTL Config Item

### DISCUSSION THREAD

**Discussion by dougblackjr** [ExpressionEngine/ExpressionEngine#3114] [Feature Requests]
Currently, sessions are hardcoded at 2 hours for a front end session and 1 hour for a CP session. This stops use cases where these should be extended or shortened.

Adding a 

In `/system/ee/legacy/libraries/Session.php`, we can change:

```
public $user_session_len = 7200;  // User sessions expire in two hours
public $cpan_session_len = 3600;  // Admin sessions expire in one hour
````

to

```
public $user_session_len = ee()->config->item('user_session_len') ? ee()->config->item('user_session_len') : 7200;
public $cpan_session_len = ee()->config->item('cpan_session_len') ? ee()->config->item('cpan_session_len') : 3600;
```

Happy to help on this!

**Comment by jaredchesebro** [comment]
I think this is worth doing. 

There is a an option for `expire_session_on_browser_close` but it's not clear what this does to session length if anything. If this setting is supposed to make session length infinite then the docs should be rewritten to reflect that.

**Comment by dougblackjr** [comment]
Added a quick PR here: https://github.com/ExpressionEngine/ExpressionEngine/pull/3115

**Reply by intoeetive** [reply]
This is getting released in 7.3
